src/util/mock/youtube.ts

```diff
@@ -0,0 +1,40 @@
+import {Page} from '@playwright/test';
+import {Mockable} from "../../testcase";
+
+export class Youtube implements Mockable{
+
+  public async mock(page: Page): Promise<void> {
+    const ytEmbedUrl = new RegExp('www\.youtube\.com', 'i');
+    await page.route(ytEmbedUrl, async route => {
+      await route.fulfill({
+        contentType: 'text/html',
+        body: `
+<html>
+<head>
+    <title>Youtube Video Mock</title>
```

Comment:
What if we screencapped a youtube video and used it in the HTML? With a title like "this is a mock video".

Reply:
We would need to do this for every service we want to mock, which might not make sense in some cases, or not provide much value.

If the goal is having more realistic snapshots than having a "red square" with some text on it, then we might find the case that we put a screenscapped video that is not appropriate, which might make things worse.
E.g a congress video on a toys e-commerce website, or a toy video on a government website.

Reply:
Makes sense. Sites can always override this!